app/controllers/charges_controller.rb

```diff
@@ -1,25 +1,50 @@
 class ChargesController < ApplicationController
 
   def new
+    @amount = (cart.total * 100)
   end
 
   def create
-  @amount = 500
+    @amount = (cart.total * 100)
 
-  customer = Stripe::Customer.create(
-    :email => 'example@stripe.com',
-    :card  => params[:stripeToken]
-  )
+    customer = Stripe::Customer.create(
+      :email => 'example@stripe.com',
+      :card  => params[:stripeToken]
+    )
 
-  charge = Stripe::Charge.create(
-    :customer    => customer.id,
-    :amount      => @amount,
-    :description => 'Rails Stripe customer',
-    :currency    => 'usd'
-  )
+    charge = Stripe::Charge.create(
+      :customer    => customer.id,
+      :amount      => @amount,
+      :description => 'Rails Stripe customer',
+      :currency    => 'usd'
+    )
 
-  rescue Stripe::CardError => e
-    flash[:error] = e.message
-    redirect_to charges_path
+    order = Crusher.create_order(session, current_user, cart)
+    session[:donations].clear
+    if current_user.phone_number
+      SendNotification.new.text_message(current_user.phone_number, order, current_user)
+    end
+    redirect_to order_path(order)
+
+    rescue Stripe::CardError => e
+      flash[:error] = e.message
+      redirect_to charges_path
+  end
+end
+
+class Crusher
+  def self.create_order(session, current_user, cart)
```

Comment:
Unused method argument - `session`. If it's necessary, use `_` or `_session` as an argument name to indicate that it won't be used.